services/ui-src/src/components/modals/AddEditReportModal.tsx

```diff
@@ -82,9 +86,29 @@ export const AddEditReportModal = ({
     setForm(customizedModalForm);
   }, [selectedReport, copyEligibleReportsByState]);
 
+  const onChange = (event: InputChangeEvent) => {
+    if (reportType === ReportType.MCPAR) {
+      // make deep copy of baseline form for customization
+      let customizedModalForm: FormJson = modalFormJson;
+
+      // user selects "Other" for the program name
+      if (
+        event.target.name === "programName" &&
+        event.target.value === "Other, specify"
+      ) {
+        setIsOtherProgramName(true);
```

Comment:
Other, specify persists the new field, and will continue to generate text fields each time you select it


https://github.com/user-attachments/assets/f0d4b485-1209-4804-a951-e094af6ecbd3


Similarly, nothing is ever setting this boolean to false, so it will be true until the user refreshes the page